pennylane/devices/default_qubit.py

```diff
@@ -46,6 +48,53 @@
 PostprocessingFn = Callable[[ResultBatch], Result_or_ResultBatch]
 
 
+observables = {
+    "PauliX",
+    "PauliY",
+    "PauliZ",
+    "Hadamard",
+    "Hermitian",
+    "Identity",
+    "Projector",
+    "SparseHamiltonian",
+    "Hamiltonian",
+    "Sum",
+    "SProd",
+    "Prod",
+    "Exp",
+    "Evolution",
+}
+
+
+def observable_stopping_coindition(obs: qml.operation.Operator) -> bool:
+    """Specifies whether or not an observable is accepted by DefaultQubit."""
+    return obs.name in observables
+
+
+def adjoint_observable_stopping_condition(obs: qml.operation.Operator) -> bool:
+    """Specifies whether or not an observable is compatible with adjoint differentiation on DefaultQubit."""
+    return obs.has_matrix
+
+
+def stopping_condition(op: qml.operation.Operator) -> bool:
+    """Specify whether or not an Operator object is supported by the device."""
+    if op.name == "QFT" and len(op.wires) >= 6:
+        return False
+    if op.name == "GroverOperator" and len(op.wires) >= 13:
+        return False
+    if op.name == "Snapshot":
+        return True
+    if op.__class__.__name__ == "Pow" and qml.operation.is_trainable(op):
+        return False
+
+    return op.has_matrix
```

Comment:
Calling this function stopping_condition makes it difficult to grasp what the function does quickly.
Also, the doc string suggests that this function will track what the device supports or not, but this is not what in fact we are using it for, not only this. We use it also to decompose, even if supported, for performance.
It would make things clearer if we changed it to decomposition_condition (or something similar) and inverted the logic. 
In this way, we make the function return true for decompose and false otherwise.

Reply:
I've used the phrasing "stopping condition" for historical reasons to draw parallel's between it and `qml.Device.stopping_condition`.

I'd be happy to rename it to something else. How about `do_decomposition` and invert the behavior?

Reply:
@tom any thoughts on the naming?

Reply:
Thanks for the suggestions! Let's stick with `stopping_condition` for now, due to historical reasons and not wanting this to be a blocker. We can revisit later if worried.

Reply:
OK for me.